### PR TITLE
Add static values to resources

### PR DESCRIPTION
**Pull request by ss2165**
* Addition and retrieval of typed ops::Const to resources
* simplify constant definitions with `KnownConstType`
* move `ConstUsize` to prelude

### CODE REVIEW

File: src/resource.rs [hunk 1]
```diff
@@ -126,6 +128,32 @@ trait TypeParametrised {
     }
 }
 
+/// A constant value provided by a resource.
+/// Must be an instance of a type available to the resource.
```
Comment (ss2165) [inline src/resource.rs:132]: maybe this should be enforced as a check on creation?

**Review by aborgna-q — APPROVED** [review]
LGTM

File: src/values.rs [hunk 2]
```diff
@@ -115,12 +116,44 @@ pub trait CustomConst:
...
 }
 
+/// Simpler trait for constant structs that have a known custom type to check against.
+pub trait KnownTypeConst: CustomConst {
+    /// The type of the constants.
+    const TYPE: CustomType;
+
+    /// Fixed implementation of [CustomConst::check_custom_type] that checks
+    /// against known correct type.
+    fn check_custom_type(&self, typ: &CustomType) -> Result<(), CustomCheckFailure> {
```
[the remark this answers is not in the document]
Comment (ss2165) [reply src/values.rs:145]: good idea yes

File: src/resource.rs (hunk 1 above)
Comment (ss2165) [reply src/resource.rs:132]: never mind - requires the resource dependency system first

**Pull request merged**